block/src/Pos/Block/Dump.hs

```diff
@@ -0,0 +1,273 @@
+{-# LANGUAGE RankNTypes #-}
+
+{- | Block dumps (used for snapshots).
+
+A block dump looks like this:
+
+@
+[block version, [block1, block2, ...]]
+@
+
+In CBOR terms, this is translated into this sequence of tokens:
+
+@
+listlen=2, block version, list_indef_len, [block1] [block2] ...
+@
+
+When we encode blocks, we strip proofs from them. When we decode blocks, we
+reconstruct those proofs.
+
+-}
+
+module Pos.Block.Dump
+       (
+       -- * Encoding
+         encodeBlockDump
+       , encodeBlockDump'
+
+       -- * Decoding
+       , decodeBlockDump
+       , decodeBlockDump'
+       , BlockDumpDecodeError (..)
+
+       -- * Internals
```

Comment:
Why are so many internals exported? E.g. where are you using `stripProof`?

Reply:
I'm not using `stripProof`, but I feel that if `BlockNoProof` is exported, then `stripProof` should be exported as well, and exporting `BlockNoProof` is needed to be able to write `Bi BlockNoProof` in type signatures.

Reply:
You have 5 exports that are never used for one `BlockNoProof`. I think this is too much.